File: src/plumbing_test/src/test02_sub_twist.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"
void doPose(const turtlesim::Pose::ConstPtr &pose)
{
    ROS_INFO("pose is (%.2f,%.2f),theta is %.2f,linearVel is %.2f,angleVel is %.2f",
    pose->x,pose->y,pose->theta,pose->linear_velocity,pose->angular_velocity);
}
int main(int argc, char  *argv[])
{
    /* code */
    ros::init(argc,argv,"sub_pose");
    ros::NodeHandle nh;
    ros::Subscriber sub=nh.subscribe("/turtle1/pose",100,doPose);
    ros::spin();
    return 0;
}

File: src/plumbing_test/scripts/test02_sub_twist.py
#! /usr/bin/env python

import rospy
from turtlesim.msg import Pose
def doPose(pose):
    rospy.loginfo("position(%.2f,%.2f),theta:%.2f,linearVel:%.2f,angularVel:%.2f",
                    pose.x,pose.y,pose.theta,pose.linear_velocity,pose.angular_velocity)
if __name__ == "__main__":
    rospy.init_node("sub_pose_p")
    sub=rospy.Subscriber("/turtle1/pose",Pose,doPose,queue_size=10)
    rospy.spin()
    pass